priv/perf/apps/load_test/lib/service/datadog/api.ex

```diff
@@ -0,0 +1,166 @@
+# limitations under the License.
+
+defmodule LoadTest.Service.Datadog.API do
+  @moduledoc """
+    Funcions for fetching monitor events from datadog.
+  """
+
+  require Logger
+
+  @datadog_events_api_path "api/v1/events"
+  @datadog_monitor_resolve_path "monitor/bulk_resolve"
+  @datadog_app_url "https://app.datadoghq.com"
+
+  def assert_metrics(environment, start_datetime, end_datetime) do
+    start_unix = DateTime.to_unix(start_datetime)
+    end_unix = DateTime.to_unix(end_datetime)
+
+    # wait for events to be emitted
+    Process.sleep(30_000)
+
+    case fetch_events(start_unix, end_unix, environment) do
+      {:ok, []} ->
+        :ok
+
+      {:ok, events} ->
+        # failed monitors with the same tags do not emit events, so we're resolving
+        # failed monitors for future runs
+        resolve_monitors(events)
+
+        {:error, events}
+
+      other ->
+        other
+    end
+  end
+
+  def fetch_events(start_time, end_time, environment) do
+    params = %{
+      start: start_time,
+      end: end_time,
+      tags: environment,
+      unaggregated: true
+    }
+
+    url = api_url() <> @datadog_events_api_path <> "?" <> URI.encode_query(params)
+
+    case HTTPoison.get(url, headers()) do
+      {:ok, %{status_code: 200, body: body}} ->
+        events =
+          body
+          |> Jason.decode!()
+          |> parse_events(environment)
+
+        {:ok, events}
+
+      {:ok, %{body: body}} ->
+        {:error, body}
+
+      {:error, error} ->
+        {:error, error}
+    end
+  end
+
+  defp parse_events(events_response, environment) do
+    events_response["events"]
+    |> Enum.filter(fn event ->
+      event["alert_type"] == "error" and String.contains?(event["text"], environment)
+    end)
+    |> Enum.map(fn event ->
+      {:ok, date} = DateTime.from_unix(event["date_happened"])
+
+      %{
+        "title" => event["title"],
+        "url" => @datadog_app_url <> event["url"],
+        "date" => date,
+        "monitor_id" => find_monitor_id(event["text"])
+      }
+    end)
+  end
+
+  defp resolve_monitors(events) do
+    params =
+      events
+      |> Enum.map(fn event -> event["monitor_id"] end)
+      |> Enum.filter(fn id -> !(is_nil(id) or id == "") end)
+      |> Enum.uniq()
+      |> Enum.map(fn id -> %{id => "ALL_GROUPS"} end)
+
+    do_resolve_monitors(params)
+  end
+
+  defp do_resolve_monitors(params, retries \\ 5)
+
+  defp do_resolve_monitors([], _), do: :ok
+
+  defp do_resolve_monitors(params, retries) do
+    payload = Jason.encode!(%{"resolve" => params})
+
+    url = api_url() <> @datadog_monitor_resolve_path
+
+    case HTTPoison.post(url, payload, headers()) do
+      {:ok, %{status_code: 200}} ->
+        :ok
+
+      {:ok, %{body: body}} ->
+        Logger.error("failed to resolve monitors #{inspect(body)}")
+
+        if retries == 0 do
+          do_resolve_monitors(params, retries - 1)
+        else
+          {:error, body}
+        end
```

Comment:
same rules apply, if/else only for functions with questionmarks
```suggestion
        case retries do
        0 ->
          do_resolve_monitors(params, retries - 1)
        _ ->
          {:error, body}
        end
```